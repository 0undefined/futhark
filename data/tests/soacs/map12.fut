// Test map in which the map function consumes its argument.
fun *[[int]] main(*[[int]] a, int i, int x) =
  map(fn *[int] (*[int] r) =>
        let r[i] = x in r,
      a)
