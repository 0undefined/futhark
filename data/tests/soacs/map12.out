[[1,1337,3], [4,1337,6]]
